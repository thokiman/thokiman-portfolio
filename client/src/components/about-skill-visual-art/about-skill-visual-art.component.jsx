import React from "react";
import { CSSTransitionGroup } from "react-transition-group";

import VisualArtGrid from "../visual-art-grid/visual-art-grid.component";
import {
  AboutVisualArtContentContainer,
  AboutVisualArtSkillContainer,
} from "./about-skill-visual-art.styles";

export const VisualArtSkill = ({ skill }) => {
  const {
    itemsPoint: { digitalArt, photography },
  } = skill;
  return (
    <AboutVisualArtSkillContainer>
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={850}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        <AboutVisualArtContentContainer>
          <VisualArtGrid visualSkill={digitalArt} />
        </AboutVisualArtContentContainer>
        <AboutVisualArtContentContainer>
          <VisualArtGrid visualSkill={photography} />
        </AboutVisualArtContentContainer>
      </CSSTransitionGroup>
    </AboutVisualArtSkillContainer>
  );
};

export default VisualArtSkill;
