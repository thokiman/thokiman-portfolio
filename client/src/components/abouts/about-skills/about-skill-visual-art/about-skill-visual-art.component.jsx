import React from 'react';
import VisualArtGrid from '../../about-commons/skill-parts/visual-art-skill-parts/visual-art-grid/visual-art-grid.component';
import {
  AboutVisualArtContentContainer,
  AboutVisualArtSkillContainer,
} from './about-skill-visual-art.styles';

export const VisualArtSkill = ({ skill }) => {
  const {
    itemsPoint: { digitalArt, photography },
  } = skill;

  return (
    <AboutVisualArtSkillContainer>
      <AboutVisualArtContentContainer>
        <VisualArtGrid visualSkill={digitalArt} />
      </AboutVisualArtContentContainer>
      <AboutVisualArtContentContainer>
        <VisualArtGrid visualSkill={photography} />
      </AboutVisualArtContentContainer>
    </AboutVisualArtSkillContainer>
  );
};

export default VisualArtSkill;
