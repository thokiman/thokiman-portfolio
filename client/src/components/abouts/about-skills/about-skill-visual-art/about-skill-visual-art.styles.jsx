import styled from 'styled-components';

const maxWidthSmartphone = 800;

export const AboutVisualArtSkillContainer = styled.div`
  width: 100%;
  height: 88%;
  overflow-y: scroll;
  overflow-x: hidden;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    display: flex;
    flex-direction: column;
    overflow-y: hidden;
    height: 100%;
  }
`;
export const AboutVisualArtContentContainer = styled.div`
  margin: 20px auto;
`;

AboutVisualArtSkillContainer.displayName = 'AboutVisualArtSkillContainer';
AboutVisualArtContentContainer.displayName = 'AboutVisualArtContentContainer';
