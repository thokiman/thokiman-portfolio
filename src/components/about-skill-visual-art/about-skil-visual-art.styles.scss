.visual-art-skill-container {

    width: 100%;
    height: 88%;
    overflow-y: scroll;
    overflow-x: hidden;

    .visual-art-content-container {
        position: relative;
        top: 5px;
        margin: 10px auto;
    }


}