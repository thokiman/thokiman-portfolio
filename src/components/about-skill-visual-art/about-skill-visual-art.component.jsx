import React from "react";
import ReactCSSTransitionGroup from "react-addons-css-transition-group";

import VisualArtGrid from "../visual-art-grid/visual-art-grid.component";
import "./about-skil-visual-art.styles.scss";

const VisualArtSkill = ({ skill }) => {
  const {
    itemsPoint: { digitalArt, photography },
  } = skill;
  return (
    <div className="visual-art-skill-container">
      <div className="visual-art-content-container">
        <VisualArtGrid visualSkill={digitalArt} />
      </div>
      <div className="visual-art-content-container">
        <VisualArtGrid visualSkill={photography} />
      </div>
    </div>
  );
};

export default VisualArtSkill;
