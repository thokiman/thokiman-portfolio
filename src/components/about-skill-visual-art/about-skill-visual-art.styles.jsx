import styled from "styled-components";

export const AboutVisualArtSkillContainer = styled.div`
  width: 100%;
  height: 88%;
  overflow-y: scroll;
  overflow-x: hidden;
`;
export const AboutVisualArtContentContainer = styled.div`
  position: relative;
  top: 5px;
  margin: 10px auto;
`;

AboutVisualArtSkillContainer.displayName = "AboutVisualArtSkillContainer";
AboutVisualArtContentContainer.displayName = "AboutVisualArtContentContainer";
