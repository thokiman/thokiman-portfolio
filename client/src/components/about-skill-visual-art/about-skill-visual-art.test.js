import React from "react";
import { shallow } from "enzyme";

import { VisualArtSkill } from "./about-skill-visual-art.component";

describe("VisualArtSkill component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      skill: {
        itemsPoint: { digitalArt: mockArray, photography: mockArray },
      },
    };
    wrapper = shallow(<VisualArtSkill {...mockProps} />);
  });

  it("should render VisualArtSkill component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
