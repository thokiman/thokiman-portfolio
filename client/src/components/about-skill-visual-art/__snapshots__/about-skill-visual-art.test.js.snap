// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualArtSkill component should render VisualArtSkill component 1`] = `ShallowWrapper {}`;
